### DISCUSSION TITLE
[Q&A] Does NVFlare support Windows 11

### DISCUSSION THREAD

**Discussion by nickbohlr** [NVIDIA/NVFlare#2142] [Q&A] [answered]
### Python version (`python3 -V`)

3.8.18

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

2.3.8

### Operating system

windows 11

### Have you successfully run any of the following examples?

- [ ] hello-numpy-sag with simulator
- [ ] hello-pt with simulator
- [ ] hello-numpy-sag with POC
- [ ] hello-pt with POC

### Please describe your question

I can successfully install nvflare on multiple machines but cannot import nvflare on any of them due to the following error:

"No module named 'resource'"

I am seeing that this is a dependency only installed on unix machines, so I'm wondering if the fact that I'm on windows and/or using miniconda is the source of my issue? I thought I'd seen others online using this library with windows, but perhaps I am mistaken?

Thank you

**Comment by YuanTingHsieh** [comment, marked as answer]
@nickbohlr We only officially support Unix/Linux currently.

One way for windows system to work you can use a "docker environment"

**Reply by AvrAmit56741** [reply]
Can you provide more guidance?

**Reply by YuanTingHsieh** [reply]
@AvrAmit56741 you can check https://docs.docker.com/desktop/install/windows-install/

And you can use the base image: nvcr.io/nvidia/pytorch:24.03-py3

There are some good materials about docker you can search for it.
Once you have docker installed and the image up and running, you can do pip install nvflare inside the docker environment and play with NVFlare.
